### PR TITLE
Added sms creator and sms sender function apps

### PR DESCRIPTION
**Pull request by vitaly-furman**

### CODE REVIEW

File: SmsCreator/function.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "bindings": [
+    {
+      "name": "smsCreatorQueue",
+      "type": "queueTrigger",
+      "direction": "in",
+      "queueName": "sms-creator-queue",
```
Comment (carmeltsadok) [inline SmsCreator/function.json:7]: change the queue name in `GenerateConferenceUrl` to `sms-creator-queue` as well.

File: SmsCreator/index.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+};
+
+const SmsCreator: AzureFunction = async (context: Context, smsData: any): Promise<void> => {
+    try {
+        await getConnection();
+
+        const smsObject = {
+            target: smsData.malshabShort.id,
+            message: getMessage(smsData.malshabShort.firstName, smsData.malshabShort.lastName, smsData.location, smsData.time),
+            sendTime: '',
```
Comment (carmeltsadok) [inline SmsCreator/index.ts:22]: why empty string and not undefined?

File: SmsSender/function.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+      "name": "smsSenderQueue",
+      "type": "queueTrigger",
+      "direction": "in",
+      "queueName": "sms-sender-queue",
+      "connection": "AzureWebJobsStorage"
+    },
+    {
+      "type": "twilioSms",
+      "name": "message",
+      "from": "+972529263458",
```
Comment (carmeltsadok) [inline SmsSender/function.json:13]: get from config

Comment (carmeltsadok) [reply SmsSender/function.json:13]: Verify if there is any way to send it from anonymous phone number (without having to write our numbers)

File: SmsSender/index.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+const SmsSender: AzureFunction = async (context: Context, smsObject: Sms): Promise<void> => {
+    try {
+        await getConnection();
+
+        const malshab = await malshabModel.findOne({ identityNumber: smsObject.target }).exec();
+        if (!malshab) {
+            throw new ObjectNotFoundFunctionError(functionName, 'malshab', smsObject.target);
+        }
+
+        await smsModel.updateOne({ target: smsObject.target }, { sendTime: new Date() });
```
Comment (carmeltsadok) [inline SmsSender/index.ts:28]: move it after the `personalPhoneNumber` validation. (because maybe it wasn't sent)

File: SmsSender/index.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+        if (!malshab.personalPhoneNumber) throw new FunctionError(functionName, 'Malshab missing personalPhoneNumber');
```
Comment (carmeltsadok) [inline SmsSender/index.ts:30]: change to validation error

File: SmsSender/index.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+        context.bindings.message = {
+            body: smsObject.message,
+            to: formatPhoneNumber('0522611936'), // TODO: in prod change to malshab.personalPhoneNumber
```
Comment (carmeltsadok) [inline SmsSender/index.ts:34]: get from config file (using environment variable that we will change on prod)

**Review by carmeltsadok — CHANGES_REQUESTED** [review]

File: SmsCreator/function.json (hunk 1 above)
Comment (AdarPascal) [reply SmsCreator/function.json:7]:
```suggestion
      "queueName": "sms-queue",
```

File: shared/sms/sms.model.ts [hunk 7]
```diff
@@ -0,0 +1,44 @@
+import { Schema, model, Document, Model, models } from 'mongoose';
+
+import { Sms } from './sms.interface';
+
+import config from '../../config';
+
```
Comment (AdarPascal) [inline shared/sms/sms.model.ts:6]: What are those blank line spaces are for?
Non Air-bnb style.
Medubar Be Hochmolog

File: SmsSender/function.json (hunk 3 above)
Comment (vitaly-furman) [reply SmsSender/function.json:13]: It's a phone number that was given by twilio

File: SmsCreator/index.ts (hunk 2 above)
Comment (vitaly-furman) [reply SmsCreator/index.ts:22]: Look in the sms model the default value is undefined

File: SmsCreator/function.json (hunk 1 above)
Comment (vitaly-furman) [reply SmsCreator/function.json:7]: queueName should remain sms-creator-queue, changed the out queue name in GenerateConferenceUrl to sms-creator-queue.

File: SmsCreator/index.ts (hunk 2 above)
Comment (vitaly-furman) [reply SmsCreator/index.ts:22]: removed the sendTime, it will be undefined by default.

File: SmsSender/function.json (hunk 3 above)
Comment (vitaly-furman) [reply SmsSender/function.json:13]: It's possible to change to Alphanumeric number, when we will know about the exact name to be sent from then will change accordingly.

File: SmsSender/index.ts (hunk 6 above)
Comment (vitaly-furman) [reply SmsSender/index.ts:34]: It's a dynamic variable, the phone number is per malshab, it should not come from enviroment variables.

**Review by carmeltsadok — APPROVED** [review]

**Review by carmeltsadok — APPROVED** [review]

**Review by AdarPascal — APPROVED** [review]

**Pull request merged**